internal/net/client_grpc.go

```diff
@@ -262,3 +250,25 @@ func (g *grpcClient) ListBeaconIDs(ctx context.Context, p Peer) (*drand.ListBeac
 	client := drand.NewPublicClient(c)
 	return client.ListBeaconIDs(context.Background(), &drand.ListBeaconIDsRequest{})
 }
+
+func (g *grpcClient) Check(ctx context.Context, p Peer) error {
+	c, err := g.conn(ctx, p)
+	if err != nil {
+		return err
+	}
+
+	client := healthgrpc.NewHealthClient(c)
+	//nolint:gomnd
+	tctx, cancel := context.WithTimeout(ctx, 3*time.Second)
```

Comment:
could move the healthcheck timeout to a field on the `grpcClient` along with its connectionTimeout